controllers/utils/utils.go

```diff
@@ -1,22 +1,66 @@
 package utils
 
 import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
 	"reflect"
 
 	kservev1beta1 "github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type IsvcDeploymentMode string
+
+var (
+	Serverless    IsvcDeploymentMode = "Serverless"
+	RawDeployment IsvcDeploymentMode = "RawDeployment"
+	ModelMesh     IsvcDeploymentMode = "ModelMesh"
 )
 
 const (
-	inferenceServiceDeploymentModeAnnotation      = "serving.kserve.io/deploymentMode"
-	inferenceServiceDeploymentModeAnnotationValue = "ModelMesh"
+	inferenceServiceDeploymentModeAnnotation = "serving.kserve.io/deploymentMode"
+	KserveConfigMapName                      = "inferenceservice-config"
 )
 
-func IsDeploymentModeForIsvcModelMesh(isvc *kservev1beta1.InferenceService) bool {
+func GetDeploymentModeForIsvc(ctx context.Context, cli client.Client, isvc *kservev1beta1.InferenceService) (IsvcDeploymentMode, error) {
+
+	// If ISVC specifically sets deployment mode using an annotation, return bool depending on value
 	value, exists := isvc.Annotations[inferenceServiceDeploymentModeAnnotation]
-	if exists && value == inferenceServiceDeploymentModeAnnotationValue {
-		return true
+	if exists {
+		switch value {
+		case string(ModelMesh):
+			return ModelMesh, nil
+		case string(Serverless):
+			return Serverless, nil
+		case string(RawDeployment):
+			return RawDeployment, nil
+		}
+	} else {
+		// ISVC does not specifically set deployment mode using an annotation, determine the default from configmap
+		controllerNs := os.Getenv("POD_NAMESPACE")
+		inferenceServiceConfigMap := &corev1.ConfigMap{}
+		err := cli.Get(ctx, client.ObjectKey{
+			Namespace: controllerNs,
+			Name:      KserveConfigMapName,
+		}, inferenceServiceConfigMap)
+		if err != nil {
+			return "", fmt.Errorf("error getting configmap 'inferenceservice-config'. %w", err)
+		}
+		var deployData map[string]interface{}
+		if err = json.Unmarshal([]byte(inferenceServiceConfigMap.Data["deploy"]), &deployData); err != nil {
+			return "", fmt.Errorf("error retrieving value for key 'deploy' from configmap %s. %w", KserveConfigMapName, err)
+		}
+		defaultDeploymentMode := deployData["defaultDeploymentMode"]
+		if defaultDeploymentMode == string(Serverless) {
```

Comment:
ModelMesh can also be a default deployment mode. See https://github.com/kserve/kserve/blob/3b15ef554aa829d0f36fec3f95a543afc4578872/config/configmap/inferenceservice.yaml#L428-L434. Let's cover that, rather than defaulting to Raw.

In fact, the defaultDeploymentMode can only have those 3 valid values (Serverless, Raw, MM). KServe code would complain if it is not set or has some other value ([ref here](https://github.com/kserve/kserve/blob/3b15ef554aa829d0f36fec3f95a543afc4578872/pkg/apis/serving/v1beta1/configmap.go#L172)). We should do the same and return an error if there is some unexpected value, rather than using a default value.

Reply:
added a switch for `defaultDeploymentMode` to handle all 3 possible options and error on any other value